ui_framework/components/card/_card_group.scss

```diff
@@ -1,23 +1,18 @@
 /**
- * 1. Use this container to offset the spacing between wrapped cards.
- */
-.kuiCardGroupContainer {
-  margin: -$cardSpacing; /* 1 */
-}
-
-/**
  * 1. Wrap cards when necessary.
+ * 2. Offset the spacing between wrapped cards.
  */
 .kuiCardGroup {
   display: flex;
   flex-wrap: wrap; /* 1 */
+  margin: -$cardSpacing; /* 2 */
 
   /**
    * 1. Use the defined width of the card to determine when to wrap.
    * 2. Use an even margin all around the card so that the spacing is still even when wrapped.
    */
   .kuiCard {
```

Comment:
Strictly speaking, this violates the open/closed principle by modifying the `.kuiCard` block based on its location inside the `.kuiCardGroup` block. I know several other frameworks break this rule by implementing such groups in exactly this way. A way to fix this would be to not nest the selectors and create a mix of `.kuiCard` and `.kuiCardGroup__card` on the individual card elements.

References:
* https://en.bem.info/methodology/css/#openclosed-principle
* https://cssguidelin.es/#the-openclosed-principle

@cjcenizal, it looks like this is consistent with how similar group blocks are implemented in the kui framework (e.g. `.kuiButtonGroup`). So this is not specific to this PR. Do we accept this BEM violation for these cases?